src/services/codefixes/convertToTypeOnlyImport.ts

```diff
@@ -30,16 +37,42 @@ registerCodeFix({
         const declaration = getDeclaration(context.sourceFile, context.span.start);
         if (declaration) {
             const changes = textChanges.ChangeTracker.with(context, t => doChange(t, context.sourceFile, declaration));
-            return [createCodeFixAction(fixId, changes, Diagnostics.Convert_to_type_only_import, fixId, Diagnostics.Convert_all_imports_not_used_as_a_value_to_type_only_imports)];
+            const importDeclarationChanges = declaration.kind === SyntaxKind.ImportSpecifier && canConvertImportDeclarationForSpecifier(declaration, context.sourceFile, context.program)
+                ? textChanges.ChangeTracker.with(context, t => doChange(t, context.sourceFile, declaration.parent.parent.parent))
+                : undefined;
+            const mainAction = createCodeFixAction(
+                fixId,
+                changes,
+                declaration.kind === SyntaxKind.ImportSpecifier
+                    ? [Diagnostics.Use_type_0, declaration.propertyName?.text ?? declaration.name.text]
+                    : Diagnostics.Use_import_type,
+                fixId,
+                Diagnostics.Fix_all_with_type_only_imports);
+
+            if (some(importDeclarationChanges)) {
+                return [
+                    createCodeFixActionWithoutFixAll(fixId, importDeclarationChanges, Diagnostics.Use_import_type),

```

Comment:
I’m not sure if the two fix options should have different `fixName`s

Reply:
Change looks good to me but i don't know about this either.

Reply:
Asked at the editor sync and this is only used for telemetry, and the telemetry stakeholders don’t care if they have the same name.